package/MDAnalysis/analysis/bat.py

```diff
@@ -0,0 +1,422 @@
+   import numpy as np
+
+   u = mda.Universe(PSF, DCD)
+
+   # selection of atomgroups
+   selected_residues = u.select_atoms("resid 5-10")
+
+   from MDAnalysis.analysis.bat import BAT
+   R = BAT(selected_residues)
+
+   # Calculate BAT coordinates for a trajectory
+   R.run()
+
+   # Reconstruct Cartesian coordinates from BAT coordinates
+   bat = R.bat[0]
+   XYZ = R.Cartesian(bat)
+
+   # The difference between the original and reconstructed coordinates
+   # should be zero.
+   print(np.sum(np.abs(XYZ - selected_residues.positions)>1E-6))
+
+After R.run(), the coordinates can be accessed with :attr:`R.bat`.
+
+
+References
+----------
+
+.. [Chang2003] Chang, Chia-En, Michael J Potter, and Michael K Gilson (2003).
+   "Calculation of Molecular Configuration Integrals". *Journal of Physical
+   Chemistry B* 107(4): 1048–55. doi:`10.1021/jp027149c
+   <https://doi.org/10.1021/jp027149c>`_
+
+.. [Hikiri2016] Hikiri, Simon, Takashi Yoshidome, and Mitsunori Ikeguchi (2016).
+   "Computational Methods for Configurational Entropy Using Internal and
+   Cartesian Coordinates." *Journal of Chemical Theory and Computation*
+   12(12): 5990–6000. doi:`10.1021/acs.jctc.6b00563
+   <https://doi.org/10.1021/acs.jctc.6b00563>`_
+
+.. [Minh2020] Minh, David D L (2020). "Alchemical Grid Dock (AlGDock): Binding
+   Free Energy Calculations between Flexible Ligands and Rigid Receptors."
+   *Journal of Computational Chemistry* 41(7): 715–30.
+   doi:`10.1002/jcc.26036 <https://doi.org/10.1002/jcc.26036>`_
+
+"""
+from __future__ import absolute_import
+# from six.moves import zip, range
+
+import numpy as np
+
+import warnings
+
+import MDAnalysis as mda
+from MDAnalysis.analysis.base import AnalysisBase
+
+from MDAnalysis.lib.distances import calc_bonds, calc_angles, calc_dihedrals
+
+
+def _sort_atoms_by_mass(atoms, reverse=False):
+    r""" Sorts a list of atoms by name and then by mass
+
+    Parameters
+    ----------
+    ag_o : list of Atoms
+      List to sort
+    reverse : bool
+      Atoms will be in descending order of mass
+
+    Returns
+    -------
+    ag_n : list of Atoms
+      Sorted list
+    """
+    return sorted(sorted(atoms, key = lambda atom:atom.name), \
+        key = lambda atom:atom.mass, reverse = reverse)
+
+
+def _find_torsion(selected_atoms, allowed_atoms):
+    """ Finds a torsion angle adjacent to the selected atoms
+
+    The torsion angle includes an atom that is allowed_atoms and is not in
+    selected_atoms.
+
+    Parameters
+    ----------
+    selected_atoms : AtomGroup
+      Atoms that have already been selected
+    allowed_atoms : AtomGroup
+      Atoms that are allowed to be part of the torsion angle
+
+    Returns
+    -------
+    new_torsion : AtomGroup
+      an AtomGroup that defines the torsion angle
+    """
+    for a1 in selected_atoms:
+        # Loop over new atoms connected to the selected atom
+        for a0 in _sort_atoms_by_mass([a for a in a1.bonded_atoms \
```

Comment:
if you feed the input to `_sort_atoms` as a generator, it will be a little more efficient.  ie you can do `_sort_atoms_by_mass(a for a in a1.bonded_atoms etc....)`

Reply:
I was also unaware of this trick.